.github/workflows/release.yaml

```diff
@@ -0,0 +1,64 @@
+name: Release
+
+on:
+    workflow_dispatch:
+        inputs:
+            new_version:
+              type: string
+            image_tag:
+              type: string
+            python_version:
+              type: string
+              default: "3.10"
+            poetry_version:
+              type: string
+              default: "1.5.1"
+
+jobs:
+    release:
+        runs-on: ubuntu-latest
+        permissions:
+            contents: write
+        steps:
+            - name: Checkout the repository
+              uses: actions/checkout@v3
+            - name: Change version in pyproject.toml
+              uses: ciiiii/toml-editor@1.0.0
+              with:
+                file: "pyproject.toml"
+                key: "tool.poetry.version"
+                value: "${{ github.event.inputs.new_version }}"
+            - name: Install Python
+              uses: actions/setup-python@v4
+              with:
+                python-version: ${{ github.event.inputs.python_version }}
+            - name: Install Poetry
+              uses: abatilo/actions-poetry@v2
+              with:
+                poetry-version: ${{ github.event.inputs.poetry_version }}
+            - name: Run poetry install
+              run: poetry install --with docs
+            - name: Run poetry build
+              run: poetry build
+            - name: Create new documentation
+              run: poetry run pdoc --html -o docs src/codeflare_sdk && pushd docs && rm -rf cluster job utils && mv codeflare_sdk/* . && rm -rf codeflare_sdk && popd
+            - name: Set Pypi token
+              run: poetry config pypi-token.pypi ${{ secrets.PYPI_TOKEN }}
+            - name: Publish new release to Pypi repository
+              run: poetry publish -r test-pypi
+            - name: Commit the pre-release changes
+              uses: stefanzweifel/git-auto-commit-action@v4
+              with:
+                commit_message: "New release: ${{ github.event.inputs.new_version }}"
+            - name: Create Github release
+              uses: ncipollo/release-action@v1
+              with:
+                  tag: "v${{ github.event.inputs.new_version }}"
+            - name: Update SDK version in the Docker image
+              run: sed -i "s/\(.*codeflare-sdk==\)[^\\s]*\(.*\)/\${{ github.event.inputs.new_version }}\2/" custom-nb-image/Dockerfile
+            - name: Update Docker image version
+              run: echo ${{ github.event.inputs.new_version }} > custom-nb-image/VERSION
```

Comment:
Based on the readme the image should be built and pushed?

Reply:
It will automatically trigger the "image" workflow by commit with changes of the VERSION file.  The "image" workflow will take care about the image build.

Reply:
Can we update that notebook image github action to be triggered on repo tag instead of the version file? @Maxusmusti wdyt?

Reply:
The approach I took for InstaScale is to trigger the action manually, tag is created as part of github action execution.